### PR TITLE
Karaf: merge master + enable feature validation

### PR DESCRIPTION
**Pull request by maggu2810**
This PR merges the current master in the karaf branch and resolves the resulted conflicts.
It also also enable the feature validation (dmx binding is still deactivated, as the current snapshots does not contain the fix, yet).

### CODE REVIEW

**Comment by maggu2810** [comment]
@dvanherbergen Sorry, I want to finish this, because I have no time to work on that this weekend.

**Comment by maggu2810** [comment]
@kaikreuzer 
After merging this PR, you can go to (after starting the runtime) "http://127.0.0.1:8181/doc/index.html" change the json URL to "http://127.0.0.1:8181/rest/swagger.json" and will see a working REST documentation.

**Comment by dvanherbergen** [comment]
@maggu2810 no problem, I'm currently busy with my day job...

File: features/org.openhab.feature.addons/src/main/feature/feature.xml [hunk 1]
```diff
@@ -0,0 +1,593 @@
...
+    </feature>
+
+    <feature name="openhab-binding-snmp" description="SNMP Binding" version="${oh1.version}">
+        <feature>openhab-runtime-base</feature>
+        <feature>openhab-runtime-compat1x</feature>
+        <bundle start-level="80">mvn:org.openhab.binding/org.openhab.binding.snmp/${oh1.version}</bundle>
+        <configfile finalname="${openhab.conf}/services/snmp.cfg" override="false">mvn:${project.groupId}/org.openhab.feature.addons.external/${project.version}/cfg/snmp</configfile>
+    </feature>
+
+    <feature name="openhab-binding-sonos" description="Sonos Binding" version="${project.version}">
```
Comment (kaikreuzer) [inline features/org.openhab.feature.addons/src/main/feature/feature.xml]: this should be esh.version

**Comment by kaikreuzer** [comment]
This PR looks a bit messy with > 80 commits, although I merged the master branch just 4 days ago the last time. Do you maybe need to rebase this branch?

**Comment by maggu2810** [comment]
I would expect a max. of 2x the commits that are done in the master. Do you think it is more?
I would prever keeping the merges as fine granular as possible until we merge this branch into master.
Then we could rebase the whole branch and get just one commit to integrate the feature 'karaf integration' into the master.

**Comment by kaikreuzer** [comment]
Well, most of the commits are older than Nov 30 (like e.g. https://github.com/maggu2810/openhab2/commit/1930370f0d27065d69a67381c46dbe9d429c1af4 from Sept 30) and that is when I last merged. So I wonder why they are included as they should be already in the Karaf branch?

**Comment by kaikreuzer** [comment]
See e.g. my last master-merge PR: https://github.com/openhab/openhab2/pull/494/commits
It only contains commits from Nov 18 - Nov 27.

**Comment by maggu2810** [comment]
I see, there must be something wrong...
I will fix this, but cannot process it before monday.

**Comment by kaikreuzer** [comment]
I've created https://github.com/openhab/openhab2/pull/523 instead - the resulting code should be identical to your branch.

**Pull request closed without merging**